src/clj/rems/ega.clj

```diff
@@ -0,0 +1,226 @@
+  `:expiration-date` - YYYY-MM-DD of the desired expiration date
+  `:reason`          - string description of the use of the key
+  `:config`          - configuration of the EGA integration"
+  [{:keys [access-token id expiration-date reason config]}]
+  (-> (http/get (str (:permission-server-url config) "/api_key/generate")
+                {:oauth-token access-token
+                 :query-params {"id" id
+                                "expiration_date" expiration-date ; TODO: pass as date?
+                                "reason" reason}
+                 :socket-timeout 2500
+                 :conn-timeout 2500})
+      :body
+      json/parse-string))
+
+(defn api-key-list
+  "List the API-Keys available.
+
+  `:access-token` - the access token
+  `:config`       - configuration of the EGA integration
+
+  NB: The token itself cannot be returned anymore."
+  [{:keys [access-token config]}]
+  (-> (http/get (str (:permission-server-url config) "/api_key")
+                {:oauth-token access-token
+                 :socket-timeout 2500
+                 :conn-timeout 2500})
+      :body
+      json/parse-string))
+
+(defn read-permissions
+  "Reads the permissions of the specified user.
+
+  `:api-key`    - valid API-Key of the person asking
+  `:account-id` - account id (Elixir or EGA)
+  `:format`     - PLAIN or JWT, defaults to JWT.
+  `:config`     - configuration of the EGA integration"
+  [{:keys [api-key account-id format config]}]
+  (-> (http/get (str (:permission-server-url config) "/permissions")
+                {:headers {"authorization" (str "api-key " api-key)
+                           "x-account-id" account-id}
+                 :query-params {"format" (or format "JWT")}
+                 :socket-timeout 2500
+                 :conn-timeout 2500})
+      :body
+      json/parse-string))
+
+(defn read-me-permissions
+  "Reads the permissions of the current user.
+
+  `:api-key`    - valid API-Key of the person asking
+  `:format`     - PLAIN or JWT, defaults to JWT.
+  `:config`     - configuration of the EGA integration"
+  [{:keys [api-key format config]}]
+  (-> (http/get (str (:permission-server-url config) "/me/permissions")
+                {:headers {"authorization" (str "api-key " api-key)}
+                 :query-params {"format" (or format "JWT")}
+                 :socket-timeout 2500
+                 :conn-timeout 2500})
+      :body
+      json/parse-string))
+
+(defn datasets-list-users
+  "Lists the users with permission to the dataset.
+
+  `:api-key`    - valid API-Key of the person asking
+  `:dataset-id` - id of the dataset
+  `:config`     - configuration of the EGA integration"
+  [{:keys [api-key dataset-id config]}]
+  (-> (http/get (str (:permission-server-url config) "/datasets/" dataset-id "/users")
+                {:headers {"authorization" (str "api-key " api-key)}
+                 :socket-timeout 2500
+                 :conn-timeout 2500})
```

Comment:
should these repeating timeouts be factored out?

Reply:
Probably makes sense. I'm not sure if they are actually needed at all or what they should be. The entitlement post used to have these but I'm not sure if they make sense?

Reply:
see also timeouts in `rems.event-notification`

Reply:
I think we'll have to see about EGA what they should eventually be. I've extracted them into local constants for now.